docs/domain/time_provider/index.md

```diff
@@ -0,0 +1,42 @@
+# Time provider
+
+`LeanCode.TimeProvider` is utilized to centralize and manage time-related operations. It enables the domain to encapsulate temporal logic, ensuring consistency and testability. Employing a dedicated time provider, facilitates easier maintenance and testing of time-dependent domain logic.
```

Comment:
That's a very extensive description of the feature that boils down to "you can't mock DateTime.Now so now we have this". :D